capi: rewrite capi image-builder for Flatcar

dongsupark opened the pull request:
Address review comments in https://github.com/kubernetes-sigs/image-builder/pull/248.

Commit:
packer: rename id_rsa to vagrant-insecure-key

To avoid confusion about the embedded rsa key, which is just a testing
key that is already available in public, we should rename `id_rsa` to a
descriptive name, `vagrant-insecure-key`.

Commit:
packer: run bootstrap from Packer instead of Ansible

When Ansible would run the Flatcar bootstrap script, it would need to
have some hacks to be able to run ansible without gather_facts, because
Flatcar does not have python installed.

To avoid such issues, instead Packer should run the Flatcar bootstrap
script.

Then we can turn on gather_facts in the Ansible part.

Commit:
Makefile: add a new target ami-flatcar

Make image-build-flatcar.sh run `make build-ami-flatcar` for Flatcar.

Commit:
packer: define ansible_os_family in Flatcar targets

Specify `ansible_os_family` to `Flatcar` only in case of Flatcar.

Commit:
packer: define containerd_cri_socket separately

`containerd_cri_socket` needs to be defined for non-Flatcar in
`packer/config/containerd.json`.
For Flatcar, it is defined in `packer/qemu/qemu-flatcar.json`,
`packer/ami/flatcar.json`, etc.

Commit:
packer: remove containerd_additional_settings

Remove unnecessary variable `containerd_additional_settings` from
`packer.json`.

Commit:
packer: remove override_system_hostname

Remove unnecessary variable `override_system_hostname`.

Commit:
packer: move boot sequences to boot_command_suffix for Flatcar

Since we need to deal with different boot sequence commands for distros,
we should define Flatcar-specific boot commands in `boot_command_prefix`,
in `packer/qemu/qemu-flatcar.json`.

Commit:
packer: run vagrant post-processors only in case of Flatcar

Run vagrant post-processors only in case of Flatcar.
Other distros do not need them.

Commit:
ansible: use jinja2 templates for containerd config.toml

Instead of having different containerd config files for Flatcar and
non-Flatcar, make use of jinja2 template in a single config file.

Commit:
ansible: make use of templates for kubelet files

For kubelet unit files as well as config files, we should use
ansible templates, to simplify the code.

For that, define new ansible variables, `systemd_prefix` and
`sysusr_prefix`.

In case of Flatcar, set kubernetes_source_type
kubernetes_cni_source_type to `http`, to make ansible download
kubernetes binaries.

Commit:
ansible: remove TasksMax from containerd.service for Flatcar

We can remove `TasksMax` from the containerd unit file,
as it is already in `containerd.service.d/max-tasks.conf`.

Commit:
ansible: remove an empty line in main.yml

Commit:
packer: set ansible_python_interpreter correctly

Specify `ansible_python_interpreter` to `/opt/bin/python` only if it is
Flatcar.

Commit:
packer: ami: split into flatcar config

Since `packer/ami/packer.json` now has common parts for all distros,
we do not need to define Flatcar-specific variables in it.
Instead, simply define necessary variables in `packer/ami/flatcar.json`.

dongsupark commented:
rebased on top of the upstream master.

Commit:
README-flatcar: use a correct name for Cluster API

vbatts commented on images/capi/packer/config/containerd.json:
```diff
@@ -2,5 +2,6 @@
     "containerd_pause_image": "k8s.gcr.io/pause:3.2",
-    "containerd_additional_settings": null
+    "containerd_additional_settings": null,
+    "containerd_cri_socket": "/var/run/containerd/containerd.sock"
```
isn't this the `/run/docker/...` socket?

dongsupark replied:
No, it needs to be `/var/run/containerd/containerd.sock`, because `images/capi/packer/config/containerd.json` is a place for default config values, mainly for non-Flatcar distros.
On the other hand, `images/capi/packer/qemu/qemu-flatcar.json` and `images/capi/packer/ami/flatcar.json` need to have configs for Flatcar, so `/run/docker/libcontainerd/docker-containerd.sock`.

vbatts commented on images/capi/packer/qemu/packer.json:
```diff
@@ -48,7 +44,7 @@
   "builders": [
     {
-      "name": "{{ user `build_name`}}",
+      "name": "flatcar-stable-2512.2.1",
```
Is this something we'll need to bump regularly upstream?

dongsupark replied:
Good point. I will soon remove the suffix `-2512.2.1`.
To do that, I would also need to update `images/capi/hack/image-build-flatcar.sh`, so it does not depend on a specific Flatcar version.

dongsupark replied:
Done

vbatts commented on images/capi/packer/qemu/packer.json:
```diff
@@ -106,5 +100,10 @@
+  "post-processors": [
+    {
+      "type": "vagrant",
+      "only": ["flatcar-stable-2512.2.1"]
```
these two also?

dongsupark replied:
Right

dongsupark replied:
Done

vbatts reviewed (approved):
It does look cleaner. Good effort.

Commit:
packer,hack: Decouple builder name from Flatcar version

Do not append suffix "-${release}" from builder names used in packer
and vagrant, so we do not need to update Flatcar version every time.

Commit:
packer: increase boot_wait only for Flatcar
